packages/grid/x-data-grid-pro/src/tests/columnPinning.DataGridPro.test.tsx

```diff
@@ -306,6 +307,7 @@ describe('<DataGridPro /> - Column pinning', () => {
         it(`should throw an error when calling \`apiRef.current.${methodName}\``, () => {
           render(<TestCase disableColumnPinning />);
           expect(() => {
+            // @ts-ignore
             apiRef.current[methodName]();
```

Comment:
I didn't find another way, and since it's for testing only I've decided to ignore it

Reply:
```ts
    (['pinColumn', 'unpinColumn', 'getPinnedColumns', 'setPinnedColumns', 'isColumnPinned'] as const).forEach(
      (methodName) => {
        it(`should throw an error when calling \`apiRef.current.${methodName}\``, () => {
          render(<TestCase disableColumnPinning />);
          expect(() => {
            (apiRef.current[methodName] as any)();
          }).to.throw();
        });
      },
    );
```

The following works
But not sure it's much better :laughing: 

I propose to remove the `forEach` which makes everything harder to read in my opinion.
And that way we assure that the test throw in caused by `disableColumnPinning` and not because the params being invalid.

```ts
    it('should throw an error when calling api methods', () => {
      render(<TestCase disableColumnPinning />);
      expect(() => apiRef.current.pinColumn('id', GridPinnedPosition.left)).to.throw();
      expect(() => apiRef.current.unpinColumn('id')).to.throw();
      expect(() => apiRef.current.getPinnedColumns()).to.throw();
      expect(() => apiRef.current.setPinnedColumns({})).to.throw();
      expect(() => apiRef.current.isColumnPinned('id')).to.throw();
    });
```

Reply:
I've removed forEach, but kept each method in separate test case, so that we know which method fails right in the test output